apps/android/app/src/main/java/chat/simplex/app/views/newchat/QRCodeScanner.kt

```diff
@@ -36,74 +37,86 @@ fun QRCodeScanner(onBarcode: (String) -> Unit) {
-          .also { it.setAnalyzer(cameraExecutor, barcodeAnalyser) }
+      }
 
-        try {
-          cameraProvider.unbindAll()
-          cameraProvider.bindToLifecycle(lifecycleOwner, cameraSelector, preview, imageAnalysis)
-        } catch (e: Exception) {
-          Log.d(TAG, "CameraPreview: ${e.localizedMessage}")
-        }
-      }, ContextCompat.getMainExecutor(context))
-    }
-  )
+      val imageAnalyzer = ImageAnalysis.Analyzer { proxy -> getQR(proxy) }
+      val imageAnalysis: ImageAnalysis = ImageAnalysis.Builder()
+        .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+        .setImageQueueDepth(1)
+        .build()
+        .also { it.setAnalyzer(cameraExecutor, imageAnalyzer) }
+      try {
+        cameraProvider.unbindAll()
+        cameraProvider.bindToLifecycle(lifecycleOwner, cameraSelector, preview, imageAnalysis)
+      } catch (e: Exception) {
+        Log.d(TAG, "CameraPreview: ${e.localizedMessage}")
+      }
+    }, ContextCompat.getMainExecutor(context))
+  }
 }
 
-class BarCodeAnalyser(
-  private val onBarcodeDetected: (barcodes: List<Barcode>) -> Unit,
-): ImageAnalysis.Analyzer {
-  private var lastAnalyzedTimeStamp = 0L
-
-  @ExperimentalGetImage
-  override fun analyze(image: ImageProxy) {
-    val currentTimestamp = System.currentTimeMillis()
-    if (currentTimestamp - lastAnalyzedTimeStamp >= TimeUnit.SECONDS.toMillis(1)) {
-      image.image?.let { imageToAnalyze ->
-        val options = BarcodeScannerOptions.Builder()
-          .setBarcodeFormats(Barcode.FORMAT_ALL_FORMATS)
-          .build()
-        val barcodeScanner = BarcodeScanning.getClient(options)
-        val imageToProcess = InputImage.fromMediaImage(imageToAnalyze, image.imageInfo.rotationDegrees)
-
-        barcodeScanner.process(imageToProcess)
-          .addOnSuccessListener { barcodes ->
-            if (barcodes.isNotEmpty()) {
-              onBarcodeDetected(barcodes)
-            } else {
-              Log.d(TAG, "BarcodeAnalyser: No barcode Scanned")
-            }
-          }
-          .addOnFailureListener { exception ->
-            Log.e(TAG, "BarcodeAnalyser: Something went wrong $exception")
-          }
-          .addOnCompleteListener {
-            image.close()
-          }
-      }
-      lastAnalyzedTimeStamp = currentTimestamp
-    } else {
-      image.close()
+private fun imageProxyToBinaryBitmap(img: ImageProxy) : BinaryBitmap? {
+  if (img.format == YUV_420_888 || img.format == YUV_422_888 || img.format == YUV_444_888) {
+    val byteBuffer: ByteBuffer = img.planes[0].buffer
+    val imageData = ByteArray(byteBuffer.capacity())
+    try {
+      byteBuffer.get(imageData)
+      val source = PlanarYUVLuminanceSource(
+        imageData,
+        img.width, img.height,
+        0, 0,
+        img.width, img.height,
+        false
+      )
+      return BinaryBitmap(HybridBinarizer(source))
+    } catch (e: Exception) {
+      e.printStackTrace()
     }
   }
+  return null
```

Comment:
Could throw here to avoid silently doing nothing on wrong format

Reply:
is it something that needs changing?

Reply:
This is no longer relevant following shift to BoofCV